### PR TITLE
Warn if "k8s_logs" config option is configured, but Kubernetes monitor is not configured / enabled

### PR DESCRIPTION
**Pull request by Kami**
This pull request updates config parsing code to warn user if ``k8s_logs`` config option is configured, but Kubernetes monitor is not enabled / configured.

## Background, Context

This "issue" was originally reported by Wei Liu. I believe he configured ``k8s_logs`` config option, but not the Kubernetes monitor itself so the ``k8s_logs`` config option had no affect.

I do think this config option is somewhat confusing - it's a top level config option, but (afaik, please correct me if I'm wrong), it's used by Kubernetes monitor so it has no affect without that monitor being configured. I assume we made it a top-level option for "simplicity" reasons, but dunno.

In other cases we don't have such issues since monitor-specific config options go under that monitor config entry.

I also asked @weilliu to confirm this and if that's indeed an issue also update documentation to make that very explicit and obvious to the end users.

### CODE REVIEW

**Comment by Kami** [comment]
I also noticed this error in the agent log when testing this so I pushed a fix in - 4abe38c1653e2c0d4455b3e19c371ebfb76fd2b0.

```bash
2021-01-20 12:34:32.710Z ERROR [core] [scalyr_logging.py:618] Failed to stop the metric log due to an exception :stack_trace:
  Traceback (most recent call last):
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/monitors_manager.py", line 201, in stop_manager
      monitor.stop(join_timeout=max_wait)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/builtin_monitors/kubernetes_monitor.py", line 4254, in stop
      self.__container_checker.stop(wait_on_join, join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/builtin_monitors/kubernetes_monitor.py", line 2788, in stop
      self.__thread.stop(wait_on_join=wait_on_join, join_timeout=join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/util.py", line 1297, in stop
      self.join(join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/util.py", line 1319, in join
      threading.Thread.join(self, timeout)
    File "/home/kami/.pythonz/pythons/CPython-3.6.9/lib/python3.6/threading.py", line 1051, in join
      raise RuntimeError("cannot join thread before it is started")
  RuntimeError: cannot join thread before it is started
```

In short, under some edge cases in Docker / Kubernetes monitor, ``.join()`` will be called before thread has been started which will result in a runtime exception.

We should ignore that error and not propagate it to the user since it's not fatal in that context.

File: scalyr_agent/util.py [hunk 1]
```diff
@@ -1316,7 +1316,12 @@ def join(self, timeout=None):
             indefinitely.
         @type timeout: float|None
         """
-        threading.Thread.join(self, timeout)
+        try:
+            threading.Thread.join(self, timeout)
+        except RuntimeError:
+            # Calling .join() on thread which is not started should not be considered fatal
```
Comment (ArthurKamalov) [inline scalyr_agent/util.py:1322]: I guess I am missing something but I cannot find the place in this PR where this error suppression may be needed.

Comment (Kami) [reply scalyr_agent/util.py:1322]: It's not related directly to this PR, but I encountered this issue multiple times when testing those changes.

In short - this edge case was there for a while already.

Comment (Kami) [reply scalyr_agent/util.py:1322]: Here is an example:

```bash
2021-01-20 12:34:32.710Z ERROR [core] [scalyr_logging.py:618] Failed to stop the metric log due to an exception :stack_trace:
  Traceback (most recent call last):
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/monitors_manager.py", line 201, in stop_manager
      monitor.stop(join_timeout=max_wait)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/builtin_monitors/kubernetes_monitor.py", line 4254, in stop
      self.__container_checker.stop(wait_on_join, join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/builtin_monitors/kubernetes_monitor.py", line 2788, in stop
      self.__thread.stop(wait_on_join=wait_on_join, join_timeout=join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/util.py", line 1297, in stop
      self.join(join_timeout)
    File "/home/kami/w/scalyr/scalyr-agent-2/scalyr_agent/util.py", line 1319, in join
      threading.Thread.join(self, timeout)
    File "/home/kami/.pythonz/pythons/CPython-3.6.9/lib/python3.6/threading.py", line 1051, in join
      raise RuntimeError("cannot join thread before it is started")
  RuntimeError: cannot join thread before it is started
```

Technically, we could also try to re-order the code so we never call ``.join()`` on a thread which has been stopped / not started yet, but that would be a larger and more risky change.

Comment (ArthurKamalov) [reply scalyr_agent/util.py:1322]: oh, sorry I haven't noticed your comment above. I think it's fine if we are sure that we do not also suppress any other errors, first of all - the timeout error. But I am pretty sure that there should be another exception type.

**Review by ArthurKamalov — APPROVED** [review]

**Pull request merged**